### PR TITLE
Fixed building with Ninja on Windows

### PR DESCRIPTION
**Pull request by Zingam**
...and updated gtest

### CODE REVIEW

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -7,20 +7,26 @@ if(MSVC)
...
 		-Dgtest_force_shared_crt=ON
+	BUILD_BYPRODUCTS
+		"<INSTALL_DIR>/lib/${CMAKE_STATIC_LIBRARY_PREFIX}gmock${CMAKE_DEBUG_POSTFIX}${CMAKE_STATIC_LIBRARY_SUFFIX}"
 )
 
 add_library(gmock UNKNOWN IMPORTED)
 set_target_properties(gmock PROPERTIES
     IMPORTED_LOCATION
-    ${PROJECT_BINARY_DIR}/test/prefix/lib/${CMAKE_STATIC_LIBRARY_PREFIX}gmock${CMAKE_STATIC_LIBRARY_SUFFIX}
+		${PROJECT_BINARY_DIR}/test/prefix/lib/${CMAKE_STATIC_LIBRARY_PREFIX}gmock${CMAKE_DEBUG_POSTFIX}${CMAKE_STATIC_LIBRARY_SUFFIX}
```
Comment (jbeder) [inline test/CMakeLists.txt]: This seems surprising. Why does it need this postfix? Why debug, given that this line isn't specific to debug?

Comment (Zingam) [reply test/CMakeLists.txt]: Gtest sets the postfix.

Comment (Zingam) [reply test/CMakeLists.txt]: For my projects I use automatic download of gtest and compilation at a configuration time but I am not sure if you'd like that.
Also the configuration and the build fail in VS2019 without the above changes.

**Comment by Zingam** [comment]
Gtest fails to compile:
`/home/travis/build/Zingam/yaml-cpp/test/gtest-1.8.1/googletest/src/gtest-typed-test.cc:109:5: error: ‘Abort’ is not a member of ‘testing::internal::posix’

**Comment by Zingam** [comment]
Looks like we need a C++11 compiler: https://github.com/google/googletest/issues/1804
Should I up the gcc version?

File: test/CMakeLists.txt (hunk 1 above)
Comment (jbeder) [reply test/CMakeLists.txt]: (I prefer to vendor the dep here.)

So gtest sets the postfix, but only for debug and ninja builds?

**Comment by jbeder** [comment]
Yes, please feel free to upgrade the compiler so that it compiles. It looks like googletest uses 4.9. That's fine with me.

**Pull request closed without merging**